upx-packer-linux.py

```diff
@@ -0,0 +1,105 @@
+#     Copyright 2019
+#
+#     Part of "Nuitka", an optimizing Python compiler that is compatible and
+#     integrates with CPython, but also works on its own.
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+#
+"""
+Linux UPX Packer and Unpacker
+
+For Packing
+Usage : python upx-packer-linux.py path/to/the/folder 
+
+For Unpacking
+Usage : python upx-packer-linux.py path/to/the/folder --decompress
+
+This script required the executable upx to be on your $PATH
+"""
+
+import os
+import subprocess as sp
+import argparse
+import time
+
+parser = argparse.ArgumentParser(
+    description="UPX-Packer of Nuitka for Linux")
+
+parser.add_argument(
+    "dir",
+    metavar = "DIR",
+    help = "The directory that is to be compressed")
+
+parser.add_argument(
+    "--decompress",
+    action = "store_true",
+    help = "decompress the UPX compresses directory")
+
+args = parser.parse_args()
+args.dir = os.path.abspath(args.dir)
+
+sep_line = "".ljust(80, "-")
+try:
+    print(sep_line)
+    print("Checking availability of UPX\n", end="", flush=True)
+    rc = sp.call(("upx", "-Vq"), stdout = sp.PIPE)               
+    print("OK: UPX is available.")
+    print(sep_line)
+except:
```

Comment:
Just as an aside, because suggested function will not raise, but return `None`, do not catch this broad an exception, this covers even `KeyboardInterrupt`, other `sys.exit`, etc. instead identify actual exceptions and catch only those, and as narrow as possible, i.e. do not include print calls in the tried block.